Use Model To Make Predictions How-To (#12038)
dotnet/docs: docs/machine-learning/how-to-guides/machine-learning-model-predictions-ml-net.md

---
title: "How-To: Use Trained Model To Make Predictions"
description: Learn to make predictions using a trained model
ms.date: 04/25/2019
author: luisquintanilla
ms.author: luquinta

> ms.custom: mvc,how-to

Reviewer: add an ms.topic: and move how-to there

> ms.custom: mvc,how-to

Reviewer: ```suggestion
ms.custom: mvc
ms.topic: how-to
```

#Customer intent: As a developer I want to use my model to make predictions
---

# How-To: Make predictions with a trained model

Learn how to use a trained model to make predictions

## Data Models

### Input Data

```csharp
public class HousingData
{
    [LoadColumn(0)]
    public float Size { get; set; }

    [LoadColumn(1, 3)]
    [VectorType(3)]
    public float[] HistoricalPrices { get; set; }

    [LoadColumn(4)]
    [ColumnName("Label")]
    public float CurrentPrice { get; set; }
}
```

### Output Data

> Like input columns names, ML.NET has default names for columns that contain the predicted value of a model. Depending on the task it may be different. 

Reviewer: input column names?

> Like input columns names, ML.NET has default names for columns that contain the predicted value of a model. Depending on the task it may be different. 

Reviewer: ```suggestion
Like input column names, ML.NET has default names for columns that contain the predicted value of a model. Depending on the task it may be different. 
```

Because the algorithm used in this sample is a linear regression algorithm, the default name of the output column is `Score` which is defined by the [`ColumnName`](xref:Microsoft.ML.Data.ColumnNameAttribute) attribute on the `PredictedPrice` property.

```csharp
class HousingPrediction : HousingData
{
    [ColumnName("Score")]
    public float PredictedPrice { get; set; }
}
```

The `HousingPrediction` data model inherits from `HousingData` to make it easy to visualize the original input data along with the output generated by the model.  

## Set up Prediction Pipeline

> Whether making a single or batch prediction, the prediction pipeline needs to be set up. After loading the data preparation pipeline and the trained model, they are combined into an [`ITransformerChain`](xref:Microsoft.ML.Data.TransformerChain%601).

Reviewer: they are -> they're

> Whether making a single or batch prediction, the prediction pipeline needs to be set up. After loading the data preparation pipeline and the trained model, they are combined into an [`ITransformerChain`](xref:Microsoft.ML.Data.TransformerChain%601).

Reviewer: ```suggestion
Whether making a single or batch prediction, the prediction pipeline needs to be set up. After loading the data preparation pipeline and the trained model, they're  combined into an [`ITransformerChain`](xref:Microsoft.ML.Data.TransformerChain%601).
```

---
title: "How-To: Use a trained model to make predictions"
description: Learn to make predictions using a trained model
ms.date: 04/29/2019
author: luisquintanilla
ms.author: luquinta
ms.custom: mvc
ms.topic: how-to
#Customer intent: As a developer I want to use my model to make predictions
---

# How-To: Make predictions with a trained model

Learn how to use a trained model to make predictions

## Data Models

### Input Data

```csharp
public class HousingData
{
    [LoadColumn(0)]
    public float Size { get; set; }

    [LoadColumn(1, 3)]
    [VectorType(3)]
    public float[] HistoricalPrices { get; set; }

    [LoadColumn(4)]
    [ColumnName("Label")]
    public float CurrentPrice { get; set; }
}
```

### Output Data

> Like input column names, ML.NET has default names for columns that contain the predicted value of a model. Depending on the task it may be different. 

Reviewer 2: >input column names, [](start = 5, length = 19)

shall we remind them which ones they are? (Label, Features, Weight)

Author: #resolved

Because the algorithm used in this sample is a linear regression algorithm, the default name of the output column is `Score` which is defined by the [`ColumnName`](xref:Microsoft.ML.Data.ColumnNameAttribute) attribute on the `PredictedPrice` property.

```csharp
class HousingPrediction : HousingData
{
    [ColumnName("Score")]
    public float PredictedPrice { get; set; }
}
```

The `HousingPrediction` data model inherits from `HousingData` to make it easy to visualize the original input data along with the output generated by the model.  

## Set up Prediction Pipeline

Whether making a single or batch prediction, the prediction pipeline needs to be set up. After loading the data preparation pipeline and the trained model, they're  combined into an [`ITransformerChain`](xref:Microsoft.ML.Data.TransformerChain%601).

```csharp
//Create MLContext 
MLContext mlContext = new MLContext();

// Load Data Preparation Pipeline and Trained Model
DataViewSchema dataPrepPipelineSchema, modelSchema;
ITransformer dataPreparationPipeline = mlContext.Model.Load("data_preparation_pipeline.zip", out modelSchema);

> ITransformer trainedModel = mlContext.Model.Load("model.zip", out modelSchema);

Reviewer 2: idk if it would sound simpler if it was only one model saved, that contained the data prep and everything

Author: #resolved

```csharp
// Create PredictionEngines
PredictionEngine<HousingData, HousingPrediction> predictionEngine = mlContext.Model.CreatePredictionEngine<HousingData, HousingPrediction>(predictionPipeline);
```

Then, use the [`Predict`](xref:Microsoft.ML.PredictionEngineBase%602.Predict*) method and pass in your input data as a parameter. Notice that using the [`Predict`](xref:Microsoft.ML.PredictionEngineBase%602.Predict*) method does not require the input to be an [`IDataView`](xref:Microsoft.ML.IDataView)). This is because it conveniently internalizes the input data type manipulation so you can pass in an object of the input data type. Additionally, since `CurrentPrice` is the target or label, don't set a value for it in the input data.

```csharp
// Input Data
HousingData inputData = new HousingData
{
    Size = 900f,
    HistoricalPrices = new float[] { 155000f, 190000f, 220000f }
};

// Get Prediction
HousingPrediction prediction = predictionEngine.Predict(inputData);
```

If you access the `Score` property of the `prediction` object, you should get a value similar to `150079`.

## Batch Prediction

> Given the following data, load it into an [`IDataView`](xref:Microsoft.ML.IDataView). Because `CurrentPrice` is the target or label, don't set a value for it in the input data.

Reviewer 2: >don't set a value [](start = 133, length = 17)

would it be better to say: "no need to set" or "we don't have a value"

Author: #resolved
